app/scripts/views/settings/avatar_camera.js

```diff
@@ -114,29 +114,42 @@ function (_, Cocktail, canvasToBlob, FormView, ProgressIndicator,
 
       this._avatarProgressIndicator = new ProgressIndicator();
       this.wrapper = this.$('#avatar-camera-wrapper');
-      this.video = this.$('#video');
+      this.video = this.$('#video')[0];
 
       this._avatarProgressIndicator.start(this.$('.progress-container'));
 
       this.canvas = this.$('#canvas')[0];
-      this.width = 320;
-      this.height = 0;
 
-      this.video[0].addEventListener('canplay', _.bind(this.canPlay, this), false);
+      this.video.addEventListener('loadedmetadata', _.bind(this.onLoadedMetaData, this), false);
```

Comment:
Switched to a more appropriate event `loadedmetadata` so we know video dimensions are set.